### ISSUE TITLE
Confusing messag when a bad SSO token is used

### ISSUE THREAD

**Issue by Kixunil** [apotdevin/thunderhub#115]
**Describe the problem/bug**
When a bad token is entered the message says something about setting up accounts instead of explaining that the token is invalid.

**Your environment**
* Version of ThunderHub: 0.8.13
* Deployment method: Debian package
* Other relevant environment details: Firefof Quantum ESR

**To Reproduce**
1. Setup ThunderHub with SSO
2. Open browser with ThunderHub URL and `token=badtoken`
3. See the message

**Expected behavior**
Error message: "Invalid SSO token"

**Actual behavior**
Message: "To start you must create an account on your server. You can find instructions for this here."

**Additional context**
BTW, `+` character in the token can't be used (URL encoding, I suppose), might be worth documenting.

**Comment by apotdevin** [comment]
Don't know if it should be shown that SSO is setup and a bad token was put in

How I had thought about it was more as an automatic process that grabs the cookie and forwards you the complete url. Didn't consider the process of copy/pasting the cookie into the url manually

What use case do you have here? Or why do you need to see that a wrong cookie was put in?

**Comment by Kixunil** [comment]
Yeah, copy-pasting is temporary in my repo until I properly implement the other side of SSO. One issue that happened recently is cookie containing those bad characters and a friend was misled into looking at that instructions on setting up account instead of understanding that the problem is somewhere else. I could imagine similar bugs in the future even if unlikely causing people to waste time.

In other words, the message should be accurate more for developers than for the users. What I learned about development during past years is that good error messages can save us lot of time.

**Comment by apotdevin** [comment]
It's a good point. I think it's better if I add some `help` notes on the main page to gives hints of what could have gone wrong.

Or do you think it's best to just show an error when a wrong cookie is used?

I dislike this idea because if a third party reaches the website I don't want them knowing that I have an sso account setup. Sort of how when you go to a `Forgot password > please input email to reset` on other websites it doesn't tell you if the email exists in their database or not.

**Comment by Kixunil** [comment]
I believe the main reasons websites don't tell that the e-mail exists are:

* Makes it harder to attack people with weak passwords
* Doesn't disclose sensitive metadata about the user being registered

Neither of those apply here (the "password is randomly generated at least 128 bits of entropy, which is unbreakable).

If you are not convinced a message saying "Something went wrong, look into the log or ask the admin to do so" and logging the error properly would still be an improvement over current misleading situation.

**Comment by apotdevin** [comment]
Done! Closed with 2de92be07461a57dab775bf384b1e9ddb001934e
